modular_skyrat/modules/modular_items/lewd_items/code/lewd_items/shibari.dm

```diff
@@ -0,0 +1,331 @@
+					them.visible_message(span_warning("[user] tied [them]'s chest!"),\
+						span_userdanger("[user] tied your chest!"),\
+						span_hear("You hear ropes being completely tightened."))
+			else
+				to_chat(user, span_warning("You don't have enough ropes!"))
+	else if(istype(them.w_uniform, /obj/item/clothing/under/shibari/groin))
+		handle_fullbody_tying(them, user)
+	else
+		to_chat(user, span_warning("They're already wearing something on this slot!"))
+
+/obj/item/stack/shibari_rope/proc/handle_arm_tying(mob/living/carbon/human/them, mob/living/user)
+	if(!(them.gloves))
+		them.visible_message(span_warning("[user] starts tying [them]'s hands!"),\
+			span_userdanger("[user] starts tying your hands!"),\
+			span_hear("You hear ropes being tightened."))
+		if(do_after(user, HAS_TRAIT(user, TRAIT_RIGGER) ? 20 : 60))
+			var/obj/item/stack/shibari_rope/split_rope = split_stack(null, 1)
+			if(split_rope)
+				shibari_hands = new(src)
+				shibari_hands.set_greyscale(greyscale_colors)
+				shibari_hands.glow = glow
+				split_rope.forceMove(shibari_hands)
+				if(them.equip_to_slot_if_possible(shibari_hands,ITEM_SLOT_GLOVES,TRUE,FALSE,TRUE))
+					shibari_hands = null
+					them.visible_message(span_warning("[user] tied [them]'s hands!"),\
+						span_userdanger("[user] tied your hands!"),\
+						span_hear("You hear ropes being completely tightened."))
+			else
+				to_chat(user, span_warning("You don't have enough ropes!"))
+	else
+		to_chat(user, span_warning("They're already wearing something on this slot!"))
+
+/obj/item/stack/shibari_rope/proc/handle_leg_tying(mob/living/carbon/human/them, mob/living/user)
+	if(!(them.shoes))
+		if(them?.dna?.mutant_bodyparts["taur"])
+			var/datum/sprite_accessory/taur/S = GLOB.sprite_accessories["taur"][them.dna.species.mutant_bodyparts["taur"][MUTANT_INDEX_NAME]]
+			if(S.hide_legs)
+				to_chat(user, span_warning("You can't tie their feet, they're a taur!"))
+				return
+		them.visible_message(span_warning("[user] starts tying [them]'s feet!"),\
+			span_userdanger("[user] starts tying your feet!"),\
+			span_hear("You hear ropes being tightened."))
+		if(do_after(user, HAS_TRAIT(user, TRAIT_RIGGER) ? 20 : 60))
+			var/obj/item/stack/shibari_rope/split_rope = split_stack(null, 1)
+			if(split_rope)
+				shibari_legs = new(src)
+				shibari_legs.set_greyscale(greyscale_colors)
+				shibari_legs.glow = glow
+				split_rope.forceMove(shibari_legs)
+				if(them.equip_to_slot_if_possible(shibari_legs,ITEM_SLOT_FEET,TRUE,FALSE,TRUE))
+					shibari_legs = null
+					them.visible_message(span_warning("[user] tied [them]'s feet!"),\
+						span_userdanger("[user] tied your feet!"),\
+						span_hear("You hear ropes being completely tightened."))
+			else
+				to_chat(user, span_warning("You don't have enough ropes!"))
+	else
+		to_chat(user, span_warning("They're already wearing something on this slot!"))
+
+/obj/item/stack/shibari_rope/proc/handle_fullbody_tying(mob/living/carbon/human/them, mob/living/user)
```

Comment:
this entire proc is copypasted like five times

Reply:
copy pasting is un-avoidable as each worn item has different behaviors on applying

Reply:
look, I don’t want to make it look like a bigger mess than it already is.

Reply:
Turning this into a proper component or element would in fact make it much less of a mess.

Reply:
Like, if you were able to componentize tying people up, then made the erp related version a subtype, we could easily add stuff like security lassos for hogtying criminals.

Reply:
we already made GAGS thing and it was really hard to implement. More components are cool but we're not doing it